DataStructureMemories/DataStructureMemories/ControllerDataStructure/DataStructureTableViewController.swift

```diff
@@ -21,20 +21,30 @@ class DataStructureTableViewController: UIViewController, UITableViewDataSource,
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         
         if let cell = tableView.dequeueReusableCell(withIdentifier: "DataStructureCell", for: indexPath) as? DataStructureCell {
+            
             cell.titleLabel.text = dataStructureManipulatios.getValueInDataSource(item: indexPath.row)
+            
             return cell
         } else {
             return tableView.dequeueReusableCell(withIdentifier: "DataStructureCell", for: indexPath)
         }
     }
-    
+
     func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
     
-    func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
         print(indexPath.row)
+        
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let detailViewController = storyboard.instantiateViewController(withIdentifier:  "DetailViewController") as! DetailViewController
+        
+        detailViewController.titleStructure = dataStructureManipulatios.getValueInDataSource(item: indexPath.row)
+        detailViewController.detailDescription = dataStructureManipulatios.descriptOfDataStructure[indexPath.row]
+        
```

Comment:
Why don't we pass whole model to the Detail controller

Reply:
fixed it